### ISSUE TITLE
Suggested change for fromBufferGeometry to include Skin information

### ISSUE THREAD

**Issue by filharvey** [mrdoob/three.js#10729]
(*** This section is for bug reports and feature requests only. This is NOT a help site. Do not ask help questions here. If you need help, please use stackoverflow. ***)

##### Description of the problem 
Skin information is not included when calling fromBufferGeometry.

`

fromBufferGeometry: function ( geometry ) {

			var scope = this;

			var indices = geometry.index !== null ? geometry.index.array : undefined;
			var attributes = geometry.attributes;

			var positions = attributes.position.array;
			var normals = attributes.normal !== undefined ? attributes.normal.array : undefined;
			var colors = attributes.color !== undefined ? attributes.color.array : undefined;
			var uvs = attributes.uv !== undefined ? attributes.uv.array : undefined;
			var uvs2 = attributes.uv2 !== undefined ? attributes.uv2.array : undefined;
			var skinIndex = attributes.skinIndex !== undefined ? attributes.skinIndex.array : undefined;
			var skinWeight = attributes.skinWeight !== undefined ? attributes.skinWeight.array : undefined;

			if ( uvs2 !== undefined ) this.faceVertexUvs[ 1 ] = [];

			var tempNormals = [];
			var tempUVs = [];
			var tempUVs2 = [];

			for ( var i = 0, j = 0, k = 0; i < positions.length; i += 3, j += 2, k += 4 ) {

				scope.vertices.push( new Vector3( positions[ i ], positions[ i + 1 ], positions[ i + 2 ] ) );

				if (skinIndex !== undefined)
				{
					scope.skinIndices.push( new Vector4( skinIndex[ k ], skinIndex[ k + 1 ], skinIndex[ k + 2 ], skinIndex[ k + 3 ] ) );
				}

				if (skinWeight !== undefined)
				{
					scope.skinWeights.push( new Vector4( skinWeight[ k ], skinWeight[ k + 1 ], skinWeight[ k + 2 ], skinWeight[ k + 3 ] ) );
				}

				if ( normals !== undefined ) {

					tempNormals.push( new Vector3( normals[ i ], normals[ i + 1 ], normals[ i + 2 ] ) );

				}

				if ( colors !== undefined ) {

					scope.colors.push( new Color( colors[ i ], colors[ i + 1 ], colors[ i + 2 ] ) );

				}

				if ( uvs !== undefined ) {

					tempUVs.push( new Vector2( uvs[ j ], uvs[ j + 1 ] ) );

				}

				if ( uvs2 !== undefined ) {

					tempUVs2.push( new Vector2( uvs2[ j ], uvs2[ j + 1 ] ) );

				}

			}

			function addFace( a, b, c, materialIndex ) {

				var vertexNormals = normals !== undefined ? [ tempNormals[ a ].clone(), tempNormals[ b ].clone(), tempNormals[ c ].clone() ] : [];
				var vertexColors = colors !== undefined ? [ scope.colors[ a ].clone(), scope.colors[ b ].clone(), scope.colors[ c ].clone() ] : [];

				var face = new Face3( a, b, c, vertexNormals, vertexColors, materialIndex );

				scope.faces.push( face );

				if ( uvs !== undefined ) {

					scope.faceVertexUvs[ 0 ].push( [ tempUVs[ a ].clone(), tempUVs[ b ].clone(), tempUVs[ c ].clone() ] );

				}

				if ( uvs2 !== undefined ) {

					scope.faceVertexUvs[ 1 ].push( [ tempUVs2[ a ].clone(), tempUVs2[ b ].clone(), tempUVs2[ c ].clone() ] );

				}

			}

			if ( indices !== undefined ) {

				var groups = geometry.groups;

				if ( groups.length > 0 ) {

					for ( var i = 0; i < groups.length; i ++ ) {

						var group = groups[ i ];

						var start = group.start;
						var count = group.count;

						for ( var j = start, jl = start + count; j < jl; j += 3 ) {

							addFace( indices[ j ], indices[ j + 1 ], indices[ j + 2 ], group.materialIndex );

						}

					}

				} else {

					for ( var i = 0; i < indices.length; i += 3 ) {

						addFace( indices[ i ], indices[ i + 1 ], indices[ i + 2 ] );

					}

				}

			} else {

				for ( var i = 0; i < positions.length / 3; i += 3 ) {

					addFace( i, i + 1, i + 2 );

				}

			}

			this.computeFaceNormals();

			if ( geometry.boundingBox !== null ) {

				this.boundingBox = geometry.boundingBox.clone();

			}

			if ( geometry.boundingSphere !== null ) {

				this.boundingSphere = geometry.boundingSphere.clone();

			}

			return this;

		},
`

##### Three.js version

- [ x] Dev
- [ ] r84
- [ ] ...

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] Linux
- [ ] Android
- [ ] IOS

**Comment by DVLP** [comment]
@Mugen87 Why closed and not merged in?

**Comment by Mugen87** [comment]
We had two PR in order to add the feature #12572. I guess it was not merged since the conversion from `BufferGeometry` to `Geometry` was intended to be temporary. I don't know if this is still true. 

From my point of view, i would not hurt to merge this one or #12572 :wink:

**Comment by DVLP** [comment]
I understand that Geometry will become soon retired but still many things are not possible on BufferGeometry. i.e. to scale the geometry(not mesh) applyMatrix will not work with BufferGeometry

This workaround is needed and works:
to Geometry => Transform => back to BufferGeometry

object.geometry = new BufferGeometry().fromGeometry(new Geometry.fromBufferGeometry(object.geometry).scale(2, 2, 2))

**Comment by Mugen87** [comment]
Why don't you use [BufferGeometry.scale](https://threejs.org/docs/#api/core/BufferGeometry.scale)?

**Comment by DVLP** [comment]
I did but it didn't work, hence the workaround

**Comment by Mugen87** [comment]
Can you show this with a [jsfiddle](https://jsfiddle.net/f2Lommf5/)? I'll take a look.

**Comment by DVLP** [comment]
It's very interesting that I can't reproduce the problem. It all works well in codepen. I used code stepping to get through all steps of scaling, creating scale matrix, applying matrix to position attributes etc. It looks like it's all working fine. Could this be "needsUpdate" problem?

I tried another workaround(below) which worked. Already no need of using old Geometry, but still I'm curious what's the root cause
`obj.geometry = new BufferGeometry().copy(obj.geometry.scale( 2, 2, 2 ));`
